crates/biome_js_semantic/src/semantic_model/model.rs

```diff
@@ -25,6 +25,32 @@ impl From<(BindingIndex, u32)> for ReferenceIndex {
     }
 }
 
+// We use `NonZeroU32` to allow niche optimizations.
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct ScopeId(std::num::NonZeroU32);
+
+// We don't implement `From<usize> for ScopeId` and `From<ScopeId> for usize`
+// to ensure that the API consumers don't create `ScopeId`.
+impl ScopeId {
+    pub(crate) fn new(index: usize) -> Self {
+        // SAFETY: We didn't handle files execedding `u32::MAX` bytes.
+        // Thus, it isn't possible to execedd `u32::MAX` scopes.
+        //
+        // Adding 1 ensurtes that the value is never equal to 0.
+        // Instead of adding 1, we could xor the value with `u32::MAX`.
+        // This is what the [nonmax](https://docs.rs/nonmax/latest/nonmax/) crate does.
+        // However, this doesn't preserve the order.
+        // It is why I opted for adding 1.
```

Comment:
Unless you put your handle at the beginning, comments should be impersonal, or use the third person